src/format_config.ml

```diff
@@ -0,0 +1,151 @@
+open! Stdune
+open Import
+open Dune_lang.Decoder
+
+let syntax =
+  Syntax.create ~name:"fmt"
+    ~desc:"integration with automatic formatters"
+    [ (1, 2) ]
+
+type language =
+  | Dialect of string
+  | Dune
+
+let language_to_dyn =
+  let open Dyn.Encoder in
+  function
+  | Dialect name -> constr "dialect" [string name]
+  | Dune -> constr "dune" []
+
+let language = function
+  | "dune" -> Dune
+  | s -> Dialect s
+
+type enabled_for =
+  | Only of language list
+  | All
+
+let languages_in_ext_1_0 =
+  [Dialect "ocaml"; Dialect "reason"]
+
+let languages_in_ext_1_1 =
+  Dune::languages_in_ext_1_0
+
+let enabled_for_field =
+ field_o "enabled_for" (repeat (map ~f:language string))
+
+let enabled_for_field_ext =
+  let+ list_opt = enabled_for_field
+  and+ ext_version = Syntax.get_exn syntax
+  in
+  match list_opt, ext_version with
+  | Some l, _ -> Only l
+  | None, (1, 0) -> Only languages_in_ext_1_0
+  | None, (1, 1) -> Only languages_in_ext_1_1
+  | None, (1, 2) -> All
+  | None, _ ->
+    Code_error.raise "This fmt version does not exist"
```

Comment:
Why is this a code_error? Can't the user see this?

Reply:
No, the only extension version is 1.2 (and there won't be any others), so these are the only possible cases. When 1.3 is present in dune-project, a generic "this version does not exist" error message is printed.